app/javascript/components/EditAccount.jsx

```diff
@@ -0,0 +1,266 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Upload, message, Form, Icon, Select, Input, Button, Slider, Switch, DatePicker, Row, Col } from 'antd';
+import "antd/dist/antd.css";
+import moment from 'moment';
+import APIRoutes from 'helpers/api_routes';
+import Utils from 'helpers/utils';
+import ActiveStorageProvider from "react-activestorage-provider";
+import '../../assets/stylesheets/application.css';
+
+
+class EditAccount extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: this.props.user,
+      email: this.props.email,
+      new_password: this.props.current_password,
+      confirm_password: null,
+      disabled: false //to prevent multiple form submissions
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleDestroy = this.handleDestroy.bind(this);
+  }
+
+  /* takes in two arrays (our array of field names and our array of values)
+   * and combines the corresponding field and value as a key value pair to be
+   * returned as a dictionary that will be sent in our JSON requests.
+   */
+  convertToDict() {
+    const user = this.state.user;
+    const keys = ["name", "address", "email", "phone", "password"];
```

Comment:
The bug was the name of the password field is actually "password" not "encrypted_password"